logicalplan/plan.go

```diff
@@ -25,6 +25,7 @@ var DefaultOptimizers = []Optimizer{
 	SortMatchers{},
 	MergeSelectsOptimizer{},
 	DetectHistogramStatsOptimizer{},
+	CoalesceOptimizer{},
```

Comment:
Should we make this opt-in first?

Reply:
If we do then we don't decide concurrently anymore - maybe it should just be a generic engine pass - we probably never don't want it

Reply:
Decode*